### ISSUE TITLE
ComboBox: User is only able to exit from the combobox after pressing 'Esc' key twice when the screen reader focus moves inside the combo box

### ISSUE THREAD

**Issue by harshika-kashyap** [microsoft/fluentui#18953]
### Environment Information

- **Package version(s)**: 8.23.3
- **Browser and OS versions**: Windows Version: Win10
Chrome version: Version 90.0.4430.212
NVDA Version: 2020.3

### Describe the issue:

User should be able to exit the ComboBox menu after interacting using 'Esc' key once while using screen reader. But here we have to press 'Esc" key two times to exit the comboBox content. People who rely on keyboard to operate the page may face difficulty while interacting with elements while NVDA is ON.

### Please provide a reproduction of the issue in a codepen: 
https://codepen.io/harshika123/pen/oNWZooz
![image](https://user-images.githubusercontent.com/26279056/125796303-26025bc9-c0ef-4e1c-a1b8-a49cf6704bc2.png)

#### Actual behavior:
User is only able to exit from the ComboBox after pressing 'Esc' key twice when the screen reader focus moves inside the combo box.

#### Expected behavior:
User should be able to exit from the ComboBox after pressing 'Esc' key only once when the screen reader focus moves inside the combo box.

**Comment by ling1726** [comment]
I think this is by design, the second `Escape` press actually telling NVDA to leave focus mode and revert to browser mode. If you decide to toggle only focus mode then the behaviour does not happen, you can do this with `INSERT+SPACEBAR` on a default NVDA install.

Here is an issue in nvda that explains the same: https://github.com/nvaccess/nvda/issues/4428
